### DISCUSSION TITLE
Suggested best practice: How to change MCU peripheral configuration (e.g., add UARTs)

### DISCUSSION THREAD

**Discussion by mws-rmain** [hathach/tinyusb#1082] [Q&A] [answered]
I'm working with a SAMD21 MCU. TinyUSB will be a key part of final functionality, but I need to make changes to the default peripheral allocation (I need more UARTs, need freedom to move SPI allocation, change GPIO pins, etc.).

The default configuration for the MCU is generated by 'atstart' (Microchip/Atmel's online configurator). What is the recommended best practice to modify this configuration?

I can generate a modified configuration by re-running atstart, but some guidance regarding which files from the new configuration are required & where they should be placed would be helpful. It looks like most will be moved to hw/mcu/microchip/samd21.

I already know there will be some integration issues (for instance, atstart will _barely_ allow USB to be enabled for the SAMD21G17, and definitely NOT if any other peripherals are added - Atmel's USB stack is so bloated), but I can work through those.

**Comment by hathach** [comment, marked as answer]
You should ask this question in the atmel forums instead.
